webhook/certificates/certificates.go

```diff
@@ -37,17 +39,27 @@ const (
 )
 
 type reconciler struct {
+	pkgreconciler.LeaderAwareFuncs
+
 	client       kubernetes.Interface
 	secretlister corelisters.SecretLister
 	secretName   string
 	serviceName  string
 }
 
 var _ controller.Reconciler = (*reconciler)(nil)
+var _ pkgreconciler.LeaderAware = (*reconciler)(nil)
 
 // Reconcile implements controller.Reconciler
 func (r *reconciler) Reconcile(ctx context.Context, key string) error {
-	return r.reconcileCertificate(ctx)
+	if r.IsLeaderFor(types.NamespacedName{
+		Namespace: system.Namespace(),
+		Name:      r.secretName,
+	}) {
+		// only reconciler the certificate when we are leader.
```

Comment:
looks like this is immutable? May be make it a member? Or perhaps it should use `key` instead?

Reply:
I moved `key` to be a member of all of the singleton reconcilers.